### PR TITLE
Disable ASCOM on Qt6

### LINKED ISSUE #2821: stellarium-1.1.1-qt6-win64.exe quits when using SharpCap
This may be a duplicate of Issue #2724 but it's possible this is not related to the Telescope Control plugin.

Expected Behaviour is for Stellarium to work when using SharpCap, just as 0.22.2 works.

Actual Behaviour is Stellarium 1.22.3 and 1.22.4 start up fine, Telescope Control plugin loads fine, Gemini Telescope.NET is selected and connected fine, and mount is slewed to the first target fine. Then as soon as I switch to SharpCap and make any mount adjustments, Stellarium quits. This can be repeated over and over.

Initially I thought this was related to the Telescope Control plugin but perhaps it is something else.

The steps described above can reproduce the problem every time.

* Stellarium version:  stellarium-1.1.1-qt6-win64.exe
* Operating system:  Windows 10 Pro
* Minisforum Celeron Mini-PC running ASCOM, Gemini Telescope.NET (latest non-beta version), Stellarium and SharpCap (latest 64-bit version).

A log file from a prior 1.22.3 session was posted in Issue #2724

### PR DESCRIPTION
**Pull request by gzotti**
### Description

We had a lot of complaints about unexpected crashes while running ASCOM-controlled telescopes.
It seems to be a conflict between ASCOM and Qt6 which we probably cannot solve. 

This branch disables ASCOM builds on Qt6. The selector radio button is greyed out, and an explanation displayed..

Other people had 2 years to help. So, disable for now, if anybody still wants to help us, you are free to do so.

In a later phase, we may consider adding support for ASCOM Alpaca, a wrapper that avoids the COM-based stuff.

Fixes #2821, #2874, 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update
- [ ] Housekeeping

### How Has This Been Tested?

**Test Configuration**:
* Operating system: Windows 11
* Graphics Card: (irrelevant)

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [x] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: guide/plg_atTheTelescope.tex [hunk 1]
```diff
@@ -921,7 +921,7 @@ \subsection{INDI}
 \item IOptron GotoNova Upgrade Kit 8400
 \end{itemize}
 
-\subsection{ASCOM (Windows only)}
+\subsection{ASCOM (Windows and Qt5-based builds only)}
```
Comment (10110111) [inline guide/plg_atTheTelescope.tex:924]: Without "and" it will be less ambiguous.

File: guide/plg_atTheTelescope.tex [hunk 2]
```diff
@@ -933,7 +933,10 @@ \subsection{ASCOM (Windows only)}
 and most of the astronomy software and tools can communicate with
 ASCOM compatible devices.
 
-The ASCOM telescope support has been added to Stellarium's Telescope Control plugin by Gion Kunz in version 0.19.3.
+The ASCOM telescope support has been added to Stellarium's Telescope Control plugin by Gion Kunz in version 0.19.3.\footnote{%
+Unfortunately ASCOM seems to be incompatible with Qt6.
```
Comment (10110111) [inline guide/plg_atTheTelescope.tex:937]: Not ASCOM, but "COM ASCOM" or "ASCOM COM". This is the name their official website uses for this Windows-specific interface (the other being Alpaca).

File: plugins/TelescopeControl/src/common/Socket.cpp [hunk 3]
```diff
@@ -40,7 +40,11 @@ long long int GetNow(void)
 		qint64 t;
 	} tmp;
 	GetSystemTimeAsFileTime(&tmp.file_time);
-	t = (tmp.t/10) - 86400000000LL*(369*365+89); // TODO: Explain this magic please!
+	// convert from NT/NTFS time epoch (1601-01-01 0:00:00 UTC) in μs to UNIX epoch (1970-01-01 0:00:00 UTC) in μs. The difference is:
+	// 24 hour/day × 3600 s/hour × ((369×365+89) day).
```
Comment (10110111) [inline plugins/TelescopeControl/src/common/Socket.cpp:44]: This misses the 10<sup>6</sup>μs/s factor that I edited into the comment a few minutes after posting it. Without it you get a million times smaller number.

File: plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp [hunk 4]
```diff
@@ -155,7 +155,12 @@ void TelescopeConfigurationDialog::createDialogContent()
 	connect(ui->radioButtonTelescopeRTS2, SIGNAL(toggled(bool)), this, SLOT(toggleTypeRTS2(bool)));
 	connect(ui->radioButtonTelescopeINDI, SIGNAL(toggled(bool)), this, SLOT(toggleTypeINDI(bool)));
 	#ifdef Q_OS_WIN
-	connect(ui->radioButtonTelescopeASCOM, SIGNAL(toggled(bool)), this, SLOT(toggleTypeASCOM(bool)));
+	#if (QT_VERSION>=QT_VERSION_CHECK(6,0,0))
+		ui->radioButtonTelescopeASCOM->setToolTip(q_("Disabled due to apparent incompatibility. Please use Qt5-based build."));
```
Comment (10110111) [inline plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp:159]: "Use Qt5-based build" is a very cryptic advice for people far from software development. At the very least it should be "Use the Qt5-based build of Stellarium". Maybe even "Use the Qt5 build of Stellarium (the one for Windows 7+)" or something like this.

File: guide/plg_atTheTelescope.tex (hunk 2 above)
Comment (gzotti) [reply guide/plg_atTheTelescope.tex:937]: OK. 20 years ago ASCOM was presented, the COM part of the name coming from Windows-specific COM. Maybe "traditional ASCOM". So far we do not mention Alpaca.

File: guide/plg_atTheTelescope.tex (hunk 1 above)
Comment (gzotti) [reply guide/plg_atTheTelescope.tex:924]: It's a simple Boolean "and". I can live with a comma instead.

File: plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp (hunk 4 above)
Comment (gzotti) [reply plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp:159]: How large shall tooltips get? Are hyperlinks to the download allowed?

Comment (alex-w) [reply plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp]: Hyperlinks will not work in tooltips

**Review by alex-w — APPROVED** [review]

File: plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp (hunk 4 above)
Comment (10110111) [reply plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp]: The tooltips can have line breaks, which will reduce their horizontal size.

Comment (gzotti) [reply plugins/TelescopeControl/src/gui/TelescopeConfigurationDialog.cpp]: There is always room for improvement...

**Pull request merged**